lxd/db/openfga/openfga.go

```diff
@@ -245,13 +332,82 @@ func (o *openfgaStore) ReadUsersetTuples(ctx context.Context, store string, filt
 
 	u, err := url.Parse(entityURL)
 	if err != nil {
-		return nil, fmt.Errorf("ReadUsersetTuples: Failed to parse entity URL %q: %w", entityURL, err)
+		return nil, err
 	}
 
-	var groupNames []string
+	cache, err := request.GetCtxValue[*RequestCache](ctx, request.CtxOpenFGARequestCache)
+	if err != nil {
+		groups, err := o.getGroupsWithEntitlementOnEntityWithURL(ctx, auth.Entitlement(filter.Relation), entityType, u)
+		if err != nil {
+			return nil, err
+		}
+
+		return storage.NewStaticTupleIterator(usersetTuples(filter.Object, filter.Relation, groups)), nil
+	}
+
+	err = o.ensureCacheLoaded(ctx, cache)
+	if err != nil {
+		return nil, err
```

Comment:
Could you please add some error wrapping?

Reply:
fixed